### PR TITLE
10X performance increase on large maps and Linux support

### PR DESCRIPTION
**Pull request by JDragan**
Nice map for testing:
https://lvlworld.com/review/id:2238

### CODE REVIEW

**Comment by JDragan** [comment]
Nice map for testing:
https://lvlworld.com/review/id:2238

File: src/InputHandlers.hpp [hunk 1]
```diff
@@ -1,13 +1,17 @@
...
+#else
+#include <SDL2/SDL.h>
+#endif
 
 /*
  * application input/event handlers
  */
 
-enum KeyCode
+enum my_KeyCode
```
Comment (kondrak) [inline src/InputHandlers.hpp]: Why the namechange? Seems unnecessary

File: src/main.cpp [hunk 2]
```diff
@@ -22,9 +22,11 @@ int main(int argc, char **argv)
 
 
     // initialize Glew
+    GLenum err;
```
Comment (kondrak) [inline src/main.cpp]: Maybe move it inside the if() so that it's in the scope only there.

File: src/main.cpp [hunk 3]
```diff
@@ -37,7 +39,7 @@ int main(int argc, char **argv)
 
     SDL_ShowCursor(SDL_DISABLE);
     g_application.OnStart(argc, argv);
-    glClearColor(0.0f, 0.0f, 0.0f, 0.0f);
+    glClearColor(0.7f, 0.0f, 0.0f, 0.0f);
```
Comment (kondrak) [inline src/main.cpp]: Leave it black please ;)

File: src/q3bsp/Q3BspLoader.hpp [hunk 4]
```diff
@@ -2,6 +2,7 @@
 #define Q3BSPLOADER_INCLUDED
 
 #include "q3bsp/Q3BspMap.hpp"
+#include <fstream>
```
Comment (kondrak) [inline src/q3bsp/Q3BspLoader.hpp]: Is this include needed here?

File: src/renderer/Font.hpp [hunk 5]
```diff
@@ -6,11 +6,11 @@
 
 class Texture;
 
-class Font
+class q3Font
```
Comment (kondrak) [inline src/renderer/Font.hpp:9]: Why the namechange?

File: src/renderer/RenderContext.cpp [hunk 6]
```diff
@@ -1,16 +1,24 @@
 #include "renderer/RenderContext.hpp"
-
+#include <iostream>
 void RenderContext::Init(const char *title, int x, int y, int w, int h)
 {
+    if (SDL_Init (SDL_INIT_EVENTS | SDL_INIT_VIDEO) < 0)
```
Comment (kondrak) [inline src/renderer/RenderContext.cpp]: SDL is initialized in main.cpp, why is this here again?

File: src/InputHandlers.hpp (hunk 1 above)
Comment (JDragan) [reply src/InputHandlers.hpp]: There is a clash with X11 KeyCode enum.
Now its changed to q3KeyCode, for consistency.

File: src/main.cpp (hunk 3 above)
Comment (JDragan) [reply src/main.cpp]: Actually I saw this, but it was too late :)

File: src/q3bsp/Q3BspLoader.hpp (hunk 4 above)
Comment (JDragan) [reply src/q3bsp/Q3BspLoader.hpp]: GCC doesn't like fstream included in the source file for some reason.
I removed from the source, now its included only in the header.

File: src/renderer/Font.hpp (hunk 5 above)
Comment (JDragan) [reply src/renderer/Font.hpp:9]: Again, X11Font name clash...

File: src/renderer/RenderContext.cpp (hunk 6 above)
Comment (JDragan) [reply src/renderer/RenderContext.cpp]: ...
